src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,95 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.dto.StationResponse;
+import nextstep.subway.exception.InvalidParameterException;
+
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "line")
+    private List<Section> sections = new ArrayList<>();
+
+    public void add(Section section) {
+        this.sections.add(section);
+    }
+
+    public void infix(Section infixSection) {
+        Section needChangedStation = getNeedChangedStationSection(infixSection);
+        needChangedStation.changed(infixSection);
+        add(infixSection);
+    }
+
+    public List<Section> getList() {
+        return sections;
+    }
+
+    private Section getNeedChangedStationSection(Section infixSection) {
+        Section changedStationSection = getInnerStationSection(infixSection);
+
+        if (changedStationSection == null) {
+            return getOuterStationSection(infixSection);
+        }
+
+        return changedStationSection;
+    }
+
+    private Section getInnerStationSection(Section infixSection) {
+        Section changedStationSection = sections
+                .stream()
+                .filter(section -> section.isUpStationEqualsId(infixSection.getUpStation().getId()))
```

Comment:
```suggestion
                .filter(section -> section.isSameUpStation(infixSection))
```
section에 private로 정의해둔 메서드를 활용하면 getter를 사용하지 않고 처리할수 있을것 같네요! 👍🏻

Reply:
앗 제가 만들어놓고...😅

Reply:
수정하고 테스트코드 돌려보니 하는 행위가 좀 다르네요 ㅠㅠ
isUpStationEqualsId() 는 section의 upstation의 id가 파라미터로 넘어온 (upstation인지 downstation인지 알수 없음) id값과 동일한건지 알아보는거고
isSameUpStation() 는 section의 upstation과 파라미터로 넘어온 section의 upstation이 동일한지 확인하네요!

Reply:
다만 getter를 좀 지양하는 방법이 있는지 한번 고민해보겠습니당!